plugins/source/gcp/client/spec.go

```diff
@@ -4,6 +4,7 @@ package client
 type Spec struct {
 	ProjectIDs            []string `json:"project_ids"`
 	ServiceAccountKeyJSON string   `json:"service_account_key_json"`
+	FolderFilter          string   `json:"folder_filter"`
 	FolderIDs             []string `json:"folder_ids"`
 	FolderRecursionDepth  *int     `json:"folder_recursion_depth"`
```

Comment:
In the future I think we should deprecate this. No need to use the recursion if we can just use a simple search query